### PR TITLE
Clean code and use enum instead of struct

### PR DESCRIPTION
**Pull request by darthpelo**

### CODE REVIEW

**Comment by benSmith1981** [comment]
What is your reason for this?

File: Yona/Yona/SignUp/SignUpFirstStepViewController.swift [hunk 1]
```diff
@@ -151,13 +151,8 @@ extension SignUpFirstStepViewController: UITextFieldDelegate {
...
     func nextTextField() {
-        if lastnameTextField.isFirstResponder() {
-            firstnameTextField.resignFirstResponder()
-            lastnameTextField.resignFirstResponder()
-        } else {
-            firstnameTextField.resignFirstResponder()
-            lastnameTextField.becomeFirstResponder()
-        }
+        firstnameTextField.resignFirstResponder()
+        lastnameTextField.isFirstResponder() ? lastnameTextField.resignFirstResponder() : lastnameTextField.becomeFirstResponder()
```
Comment (benSmith1981) [inline Yona/Yona/SignUp/SignUpFirstStepViewController.swift]: Bit more verbose I see!

Comment (darthpelo) [inline Yona/Yona/SignUp/SignUpFirstStepViewController.swift]: I don't like repletion of code in if statement. In this case `firstnameTextField.resignFirstResponder()` is called always and then the `?:` is more useful in this case instead when the result of the check is only on instruction.

File: Yona/Yona/SignUp/SignUpFirstStepViewController.swift [hunk 2]
```diff
@@ -152,7 +152,8 @@ extension SignUpFirstStepViewController: UITextFieldDelegate {
     
     func nextTextField() {
         firstnameTextField.resignFirstResponder()
-        lastnameTextField.isFirstResponder() ? lastnameTextField.resignFirstResponder() : lastnameTextField.becomeFirstResponder()
+        if lastnameTextField.isFirstResponder() { lastnameTextField.resignFirstResponder() }
```
Comment (darthpelo) [inline Yona/Yona/SignUp/SignUpFirstStepViewController.swift]: Small refactor

**Pull request merged**